[doc] Improve Pulsar Administration load-balance (#5265)
apache/pulsar: site2/docs/administration-load-balance.md

The stateless nature of brokers makes the dynamic assignment possible, so you can quickly expand or shrink the cluster based on usage.

#### Assignment granularity

The assignment of topics or partitions to brokers is not done at the topics or partitions level, but done at the Bundle level (a higher level). The reason is to amortize the amount of information that you need to keep track. Based on CPU, memory, traffic load and other indexes, topics are assigned to a particular broker dynamically. 

Instead of individual topic or partition assignment, each broker takes ownership of a subset of the topics for a namespace. This subset is called a "*bundle*" and effectively this subset is a sharding mechanism.

The namespace is the "administrative" unit: many config knobs or operations are done at the namespace level.

For assignment, a namespaces is sharded into a list of "bundles", with each bundle comprising
a portion of overall hash range of the namespace.

> You can take the hash of the topic name and see in which bundle the hash falls into to assign topics to a particular bundle.

Reviewer: Using "you" here makes it sounds as an operation that will be done by a user/administrator, while in fact this is done by brokers.

## Load balance across Pulsar brokers

Pulsar is an horizontally scalable messaging system, so the traffic
in a logical cluster must be spread across all the available Pulsar brokers as evenly as possible, which is a core requirement.

> In most cases, this requirement is true out of the box and you should not worry about it. Though,you can use multiple settings and tools to control the traffic distribution and they require a bit of context to understand how the traffic is managed in Pulsar.

Reviewer 2: refine this sentence, pay attention to punctuation and try to make it easy to read and understand.

> ## Pulsar load manager architecture

Reviewer 2: Do we need to add a lead-in sentence between heading and its subheading?

### Assignment granularity

The assignment of topics or partitions to brokers is not done at the topics or partitions level, but done at the Bundle level (a higher level). The reason is to amortize the amount of information that you need to keep track. Based on CPU, memory, traffic load and other indexes, topics are assigned to a particular broker dynamically. 

Instead of individual topic or partition assignment, each broker takes ownership of a subset of the
topics for a namespace. This subset is called a "*bundle*" and effectively this subset is a sharding
mechanism.

The namespace is the "administrative" unit: many config knobs or operations are done at the namespace level.

For assignment, a namespaces is sharded into a list of "bundles", with each bundle comprising
a portion of overall hash range of the namespace.

You can take the hash of the topic name and see in which bundle the hash falls into to assign topics to a particular bundle.

Each bundle is independent of the others and thus is independently assigned to different brokers.

### Create namespaces and bundles

> When you create a new namespace, the new namrspace sets to use the default number of bundles. You can set this in `conf/broker.conf`:

Reviewer 2: namrspace?

```properties
# When a namespace is created without specifying the number of bundle, this
# value will be used as the default
defaultNumberOfNamespaceBundles=4
```

You can either change the system default, or override it when you create a new namespace:

```shell
$ bin/pulsar-admin namespaces create my-tenant/my-namespace --clusters us-west --bundles 16
```

With this command, you create a namespace with 16 initial bundles. Therefore the topics for this namespaces can immediately be spread across up to 16 brokers.

> In general, if the expected traffic and number of topics is known in advance, to start with a reasonable number of bundles instead of waiting for the system to auto-correct the

Reviewer 2: You can further improve the sentence.

distribution is a good idea.

> On a same note, to start with more bundles than number of brokers is normally beneficial, primarily because of the hashing nature of the distribution of topics into bundles. For example, for a namespace with 1000 topics, using something like 64 bundles achieves a good distribution

Reviewer 2: Refine the first sentence.

of traffic across 16 brokers.

### Unload topics and bundles

> You can use an admin operation of "unloading" a topic in Pulsar. Unloading means to close the topics,

Reviewer 2: ```suggestion
You can "unload" a topic in Pulsar with admin operation. Unloading means to close the topics,
```

pulsar-admin namespaces unload tenant/namespace
```

### Split namespace bundles 

Since the load for the topics in a bundle might change over time, or could just be hard to predict

> upfront, brokers can spilt bundles  in 2. The new smaller bundles can then be reassigned to different brokers.

Reviewer 2: check this sentence again.
